.github/workflows/nightly.yml

```diff
@@ -389,9 +389,47 @@ jobs:
           echo opcache.jit_hot_side_exit=1 >> /etc/php.d/opcache.ini
           echo memory_limit=-1 >> /etc/php.d/opcache.ini
           php -v
+      - name: Test PHPSeclib
+        if: always()
+        run: |
+          git clone https://github.com/phpseclib/phpseclib --branch=master
+          cd phpseclib
+          git checkout 259bd9f1e8af11726ed74acf527c2c046549061b
+          export ASAN_OPTIONS=exitcode=139
+          export PHPSECLIB_ALLOW_JIT=1
+          php /usr/bin/composer install --no-progress --ignore-platform-reqs
+          php vendor/bin/paratest --verbose --configuration=tests/phpunit.xml --runner=WrapperRunner || exit $?
+          git clean -ffdx
+      - name: Test Psalm
+        if: matrix.branch.ref != 'PHP-8.0'
+        run: |
+          git clone https://github.com/vimeo/psalm --branch=master
+          cd psalm
+          git checkout 7428e49b115a2a837aa29cf0fafd0ca902fe2457
+          export ASAN_OPTIONS=exitcode=139
+          export PSALM_ALLOW_XDEBUG=1
+          php /usr/bin/composer install --no-progress --ignore-platform-reqs
+          php ./psalm --no-cache || exit $?
+      - name: Test PHPStan
+        if: matrix.branch.ref != 'PHP-8.0'
+        run: |
+          git clone https://github.com/phpstan/phpstan-src
+          cd phpstan-src
+          git checkout 1f608dc6a560f4a5accc854add8e005da0e7ceea
+          sed 's/80299/89999/g' -i conf/config.neon
```

Comment:
A more generic regex might be able to avoid having to bump the version continuously.